src/pages/TermsOfService.ts

```diff
@@ -0,0 +1,148 @@
+import _ from 'lodash/fp';
+import { useState } from 'react';
+import { div, h, h1, img, WithKey } from 'react-hyperscript-helpers';
+import { ButtonOutline, ButtonPrimary, ButtonSecondary } from 'src/components/common';
+import { centeredSpinner } from 'src/components/icons';
+import { MarkdownViewer, newWindowLinkRenderer } from 'src/components/markdown';
+import scienceBackground from 'src/images/science-background.jpg';
+import { Ajax } from 'src/libs/ajax';
+import { signOut } from 'src/libs/auth';
+import colors from 'src/libs/colors';
+import { reportError, withErrorReporting } from 'src/libs/error';
+import * as Nav from 'src/libs/nav';
+import { useOnMount } from 'src/libs/react-utils';
+import { authStore } from 'src/libs/state';
+import * as Style from 'src/libs/style';
+import * as Utils from 'src/libs/utils';
+
+const TermsOfServicePage = () => {
+  const [busy, setBusy] = useState<boolean>();
+  const { isSignedIn, termsOfService } = authStore.get(); // can't change while viewing this without causing it to unmount, so doesn't need to subscribe
+  const acceptedLatestTos = isSignedIn && termsOfService.userHasAcceptedLatestTos;
+  const usageAllowed = isSignedIn && termsOfService.permitsSystemUsage;
+  const [tosText, setTosText] = useState();
+
+  useOnMount(() => {
+    const loadTosAndUpdateState = _.flow(
+      Utils.withBusyState(setBusy),
+      withErrorReporting('There was an error retrieving our terms of service.')
+    )(async () => {
+      setTosText(await Ajax().User.getTos());
+    });
+    loadTosAndUpdateState();
+  });
+
+  const accept = async () => {
+    try {
+      setBusy(true);
+      const { enabled } = await Ajax().User.acceptTos();
+      const termsOfService = await Ajax().User.getTermsOfServiceComplianceStatus();
+
+      if (enabled) {
+        const registrationStatus = 'registered';
+        authStore.update((state) => ({ ...state, registrationStatus, termsOfService }));
+        Nav.goToPath('root');
+      } else {
+        reportError('Error accepting TOS, unexpected backend error occurred.', new Error('Cannot accept ToS'));
```

Comment:
Alternatively, throwing an error here would cause it to be reported by the `reportError` in the `catch` block.

```suggestion
        throw new Error('Cannot accept ToS'));
```

That would avoid having to duplicate error reporting code here and in the `catch` block.